repos/system_upgrade/el7toel8/libraries/requirearchver.py

```diff
@@ -0,0 +1,130 @@
+
+    detected = platform.machine()
+    reason = 'architecture'
+    if detected not in required:
+        _log_and_raise(reason, required, detected)
+
+    _log_passed(reason, required, detected)
+
+
+def _require_minor_version(destination, required, detected):
+    """
+    Stop actor's execution if requirement on minor version is not met.
+
+    :param destination: 'source' or 'target'
+    :type destination: string
+    :param required: list or tuple of versions - integers or strings in the form '<integer>['+'|'>'|'-'|'<']'
+    :type required: list, tuple
+    :param detected: Detected version
+    :type detected: int
+    """
+
+    reason = "{}'s minor version".format(destination)
+
+    if not isinstance(required, (list, tuple)):
+        raise TypeError("Required {reason} has to be a list or tuple: {required}".format(
+            reason=reason, required=required))
+
+    if all(isinstance(r, int) for r in required):
+        # required is list of versions
+        if detected not in required:
+            _log_and_raise(reason, required, detected)
+    else:
+        # required should be list of '<integer>['+'|'>','-','<']' strings
+        fun_map = {'+': operator.gt, '>': operator.gt, '-': operator.lt, '<': operator.lt}
+        try:
+            if not all(fun_map[r[-1]](detected, int(r[:-1])) for r in required):
+                _log_and_raise(reason, required, detected)
+        except (KeyError, ValueError, TypeError):
+            raise TypeError("Required {reason} has to be a list or tuple of integers or strings in the "
+                            "form '<integer>['+'|'>'|'-'|'<']': {required}".format(
+                                reason=reason, required=required))
+
+    _log_passed(reason, required, detected)
+
+
+# TODO: target's minor version will be specified using dynamic configuration phase, see #505 (add tests afterward)
+# def require_tgt_minor_version(required):
+#     """
+#     Stop actor's execution if requirement on minor version of target system is not met.
+#
+#     :param destination: 'source' or 'target'
+#     :type destination: string
+#     :param required: list or tuple of versions - integers or strings in the form '<integer>['+'|'>'|'-'|'<']'
+#     :type required: list, tuple
+#     :param detected: Detected version
+#     :type detected: int
+#     """
+#     detected = 0
+#     _require_minor_version('target', required, detected)
+
+
+def require_src_minor_version(required):
+    """
+    Stop actor's execution if requirement on minor version of source system is not met.
+
+    :param destination: 'source' or 'target'
+    :type destination: string
+    :param required: list or tuple of versions - integers or strings in the form '<integer>['+'|'>'|'-'|'<']'
+    :type required: list, tuple
+    :param detected: Detected version
+    :type detected: int
+    """
+    detected = int(get_os_release_info('/etc/os-release')['VERSION_ID'].split('.')[1])
```

Comment:
You have to use data from the actor config (PR is waiting for review and finishing) or you have to use OSReleaseFacts as temporary workround. You cannot access the file in all phases.